tests/cecil-tests/AttributeTest.cs

```diff
@@ -0,0 +1,257 @@
+// 				parentAvailability.Append("maccatalyst");
+// 			}
+
+			var myAvailability = GetAvailabilityAttributes (item, includeUnsupported: true);
+			if (!FirstContainsAllOfSecond (myAvailability, parentAvailability)) {
+				DebugPrint (fullName, parentAvailability, myAvailability);
+				found.Add (fullName);
+			}
+		}
+
+		// https://github.com/xamarin/xamarin-macios/issues/10170
+		// Every binding class that has net6 any availability attributes on a method/property
+		// must have an introduced for the current platform.
+		//
+		// Example:
+		// class TestType
+		// {
+		//     public static void Original () { }
+		//
+		//     [SupportedOSPlatform(""ios2.0"")]
+		//     public static void Extension () { }
+		// }
+		//
+		// When run against mac, this fails as Extension does not include a mac supported of any kind attribute
+		// [TestCaseSource (typeof (Helper), "NetPlatformAssemblies")]
+		public void AllAttributedItemsMostIncludeCurrentPlatform (string assemblyPath)
+		{
+			var assembly = Helper.GetAssembly (assemblyPath);
+			if (assembly is null) {
+				Assert.Ignore ("{assemblyPath} could not be found (might be disabled in build)");
+				return;
+			}
+
+#if DEBUG
+			Console.WriteLine(assemblyPath);
+#endif
+
+			string platformName = AssemblyToAttributeName (assemblyPath);
+
+			HashSet<string> found = new HashSet<string> ();
+			foreach (var type in Helper.FilterTypes (assembly, a => HasAnyAvailabilityAttribute (a, includeUnsupported: true))) {
+				CheckCurrentPlatformIncludedIfAny (type, platformName, type.FullName, type.DeclaringType, found);
+			}
+			foreach (var prop in Helper.FilterProperties (assembly, a => HasAnyAvailabilityAttribute (a, includeUnsupported: true))) {
+				CheckCurrentPlatformIncludedIfAny (prop, platformName, prop.FullName, prop.DeclaringType, found);
+			}
+			foreach (var meth in Helper.FilterMethods (assembly, a => HasAnyAvailabilityAttribute (a, includeUnsupported: true))) {
+				CheckCurrentPlatformIncludedIfAny (meth, platformName, meth.FullName, meth.DeclaringType, found);
+			}
+
+			Assert.That (found, Is.Empty);
+		}
+
+		void CheckCurrentPlatformIncludedIfAny (ICustomAttributeProvider item, string platformName, string fullName, TypeDefinition parent, HashSet<string> found)
+		{
+			if (HasAnyAvailabilityAttribute (item, true)) {
+				var supportedAttributes = item.CustomAttributes.Where (a => IsSupportedAttribute (a));
+				if (!supportedAttributes.Any (a => FindAvailabilityKind (a) != platformName)) {
+					found.Add (fullName);
+				}
+			}
+		}
+
+		string AssemblyToAttributeName (string assemblyPath)
+		{
+			switch (Path.GetFileName (assemblyPath)) {
+			case "Xamarin.iOS.dll":
+				return "ios";
+			case "Xamarin.TVOS.dll":
+				return "tvos";
+			case "Xamarin.Mac.dll":
+				return "macos";
+			case "Xamarin.MacCatalyst.dll":
+				return "maccatalyst";
+			default:
+				throw new NotImplementedException ();
```

Comment:
The assembly names will change soon, so implementing it this way makes it not require any changes when that happens:

```suggestion
			var baseName = Path.GetFileName (assemblyPath));
			if (Configuration.GetBaseLibraryName (TargetFramework.DotNet_5_0_iOS) == baseName)
				return "ios";
			if (Configuration.GetBaseLibraryName (TargetFramework.DotNet_5_0_tvOS) == baseName)
				return "tvos";
			if (Configuration.GetBaseLibraryName (TargetFramework.DotNet_5_0_macOS) == baseName)
				return "macos";
			if (Configuration.GetBaseLibraryName (TargetFramework.DotNet_5_0_MacCatalyst) == baseName)
				return "maccatalyst";
			throw new NotImplementedException ();
```